src/util/motionModel.ts

```diff
@@ -298,6 +307,55 @@ let emptyIterationCounter = 0;
 let prevGnssFile = '';
 let prevGpsRecord: GNSS | undefined = undefined;
 
+const createFileNameForFAT32 = (gnssFile: string, fileCreationDate: string) => {
+  //We exclude all dots and colons from the filename and replace with - for FAT32 compatibility
+  const parts = gnssFile.split('.');
+  if (parts.length > 1) {
+    const lastPart = parts.pop(); // Remove the last part
+    const replacedString = parts.join('-') + '.' + lastPart;
+    return fileCreationDate + '/' + replacedString;
+  }
+  return '';
+};
+
+const copyFileToUSB = async (fileName: string, sourceFolder: string, fileType: FILETYPE) => {
+
+  const execAsync = promisify(exec);
+
+  // Replace all colons and periods with dashes to make them compatible with FAT32
+  const fileNameForFAT32 = fileName.split('/').pop()?.replace(/:/g, '-');
+
+  if (fileNameForFAT32) {
+    // Get filename of GNSS file
+
+    const usbConnected = existsSync(USB_WRITE_PATH);
+
+    if (usbConnected) {
+
+      const fileCreationDate = fileNameForFAT32.split('T')[0];
+
+      const destinationFileName = createFileNameForFAT32(fileNameForFAT32, fileCreationDate);
+
+      if (destinationFileName) {
+        const destinationFilePath = sourceFolder + '/' + destinationFileName;
+
+        try {
+          await fs.mkdirSync(path.join(sourceFolder, fileCreationDate));
```

Comment:
@HariHive do you mind creating a sub-folder for every file type?
so it will be 
/2023-10-10
  /gps
  /imu
  
I think it will be slightly more clean to put it in subfolders, otherwise the day folder itself will still be looking pretty messy.